fhd_core/gridding/visibility_degrid.pro

```diff
@@ -102,8 +102,8 @@ dx1dy1_arr = Temporary(dx_arr) * Temporary(dy_arr)
 xmin=Long(Floor(Temporary(xcen))+dimension/2-(psf_dim/2-1))
 ymin=Long(Floor(Temporary(ycen))+elements/2-(psf_dim/2-1))
 
-range_test_x_i=where((xmin LE 0) OR ((xmin+psf_dim-1) GE dimension-1),n_test_x)
-range_test_y_i=where((ymin LE 0) OR ((ymin+psf_dim-1) GE elements-1),n_test_y)
+range_test_x_i=where(((xmin LE 0) OR ((xmin+psf_dim-1) GE dimension-1),n_test_x)
+range_test_y_i=where(((ymin+(psf_dim/2-1)) LE 0) OR ((ymin+psf_dim-1) GE elements-1),n_test_y)
```

Comment:
I think the existing code is actually OK. It tests whether any pixel of the psf for the given baseline would fall out of the UV plane, and we do want to exclude anything that is outside. Note that xmin and ymin are defined above to be the minimum x and y values of the box of pixels with nonzero values, and are not the coordinates of the baseline. Also note that in this coordinate system baselines with zero length are in the center of the plane, so they should not be affected by this test.